modules/blockbuilder/util/id.go

```diff
@@ -21,43 +21,32 @@ type IDGenerator interface {
 var _ IDGenerator = (*DeterministicIDGenerator)(nil)
 
 type DeterministicIDGenerator struct {
-	seeds []int64
-	seq   *atomic.Int64
+	tenantBytes []byte
+	seeds       []int64
+	seq         *atomic.Int64
 }
 
 func NewDeterministicIDGenerator(tenantID string, seeds ...int64) *DeterministicIDGenerator {
-	seeds = append(seeds, int64(binary.LittleEndian.Uint64(stringToBytes(tenantID))))
 	return &DeterministicIDGenerator{
-		seeds: seeds,
-		seq:   atomic.NewInt64(0),
+		tenantBytes: []byte(tenantID),
+		seeds:       seeds,
+		seq:         atomic.NewInt64(0),
 	}
 }
 
 func (d *DeterministicIDGenerator) NewID() backend.UUID {
 	seq := d.seq.Inc()
-	seeds := append(d.seeds, seq)
-	return backend.UUID(newDeterministicID(seeds))
+	return backend.UUID(newDeterministicID(d.tenantBytes, append(d.seeds, seq)))
 }
 
-func newDeterministicID(seeds []int64) uuid.UUID {
-	b := int64ToBytes(seeds...)
+func newDeterministicID(b []byte, seeds []int64) uuid.UUID {
+	sl, dl := len(seeds), len(b)
+	data := make([]byte, dl+sl*8) // 8 bytes per int64
+	copy(b, data)
 
-	return uuid.NewHash(hash, ns, b, 5)
-}
-
-// TODO - Try to avoid allocs here
-func stringToBytes(s string) []byte {
-	return []byte(s)
-}
-
-func int64ToBytes(seeds ...int64) []byte {
-	l := len(seeds)
-	bytes := make([]byte, l*8)
-
-	// Use binary.LittleEndian or binary.BigEndian depending on your requirement
 	for i, seed := range seeds {
-		binary.LittleEndian.PutUint64(bytes[i*8:], uint64(seed))
+		binary.LittleEndian.PutUint64(data[dl+i*8:], uint64(seed))
 	}
 
-	return bytes
+	return uuid.NewHash(hash, ns, data, 5)
```

Comment:
It's so hash.Hash is not created each time, saving some allocations.

Reply:
The current global var isn't safe for concurrent use, how about making it a struct var?  I know the ID generator isn't called concurrently yet, but there's nothing preventing it, and seems likely (parallelism in the block builder).

Reply:
I've been strongly preferring the `fnv1a` static methods that require no struct and have no concurrency concerns.

https://pkg.go.dev/github.com/segmentio/fasthash/fnv1a

Reply:
>The current global var isn't safe for concurrent use, how about making it a struct var?

Moved the `hash.Hash` to a struct field. Agree it's safer, and the difference won't be noticeable.

>I've been strongly preferring the fnv1a static methods that require no struct and have no concurrency concerns.
https://pkg.go.dev/github.com/segmentio/fasthash/fnv1a

From what I (read)[https://github.com/segmentio/fasthash?tab=readme-ov-file#motivations], the main arguments for using fasthash is saving allocs and avoiding inefficient string to bytes conversions.

We'd need to do the alloc manually since we're reusing the data byte slice between sequential IDs and already have bytes. I don't see a big benefit on using other than Go's sha1 in this case.

Reply:
Yes agree that would be my concern with fnv.  I think we can start with the SHAs in the uuid package and go from there.  Realistically this is not called much (once per block), much less than trace/span hashing.